src/app/components/services/query-builder/queryparser.service.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+/**
+ * @memberof transmartBaseUi
+ * @ngdoc factory
+ * @name QueryParserService
+ */
+angular.module('transmartBaseUi').factory('QueryParserService', ['XML2JSONService', 'StudyListService', 'TreeNodeService', 'PromiseQueue',
+    function (XML2JSONService, StudyListService, TreeNodeService, PromiseQueue) {
+
+        var service = {};
+
+      /**
+       * Puts the variable in a list if it is not already a list.
+       * @param variable The variable to return as a list
+       * @returns {*[]} The original list or the variable put in the list
+       */
+        function makeList(variable) {
+            return variable.constructor === Array ? variable : [variable];
```

Comment:
I think I based it on this: http://stackoverflow.com/a/26633883/1097971
Do you want me to change it?